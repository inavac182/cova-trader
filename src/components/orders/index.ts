export * from './Orders';
export * from './Title';
export * from './OrdersList';
