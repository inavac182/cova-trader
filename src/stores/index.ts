export * from './home-page.store';
export * from './books.store';
export * from './ticker.store';
