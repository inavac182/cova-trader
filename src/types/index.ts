export * from './books-types';
export * from './ticker-types';
export * from './asks-types';
export * from './bids-types';
export * from './orders-types';
export * from './match-types';
