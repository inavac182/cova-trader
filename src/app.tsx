import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';

import { RouterStore, syncHistoryWithStore } from 'mobx-react-router';
import { Provider } from 'mobx-react';

import { HomePageStore, BooksStore, TickerStore } from './stores';
import { HomePage } from './components';
import './less/main.less';
import { OrdersStore } from './stores/orders.store';

const browserHistory = createBrowserHistory();
const routingStore = new RouterStore();

const stores = {
  routing: routingStore,
  homePageStore: new HomePageStore(),
  booksStore: new BooksStore(),
  tickerStore: new TickerStore(),
  ordersStore: new OrdersStore(),
};

const history = syncHistoryWithStore(browserHistory, routingStore);

export default function MainRouter() {
  return (
    <Provider {...stores}>
      <Router history={history}>
        <HomePage />
      </Router>
    </Provider>
  );
}

ReactDOM.render(<MainRouter />, document.getElementById('main'));
