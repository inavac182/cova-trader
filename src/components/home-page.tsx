import * as React from 'react';
import { observer, inject } from 'mobx-react';
import { HomePageStore } from 'src/models';

export interface HomePageProps {
  homePageStore?: HomePageStore;
}

export const HomePage = inject('homePageStore')(
  observer((props: HomePageProps) => {
    const { homePageStore } = props;
    const { title } = homePageStore;
    const [titleState, setTitle] = React.useState(title);

    const handleOnChange = e => {
      setTitle(e.target.value);
    };

    const handleSubmit = e => {
      homePageStore.hydrate({
        title: titleState,
      });

      e.preventDefault();
    };

    return (
      <div className="page">
        <h1>{title}</h1>
        <form onSubmit={handleSubmit}>
          <input type="text" value={titleState} onChange={handleOnChange} required />
          <button type="submit"> Save! </button>
        </form>
      </div>
    );
  })
);
