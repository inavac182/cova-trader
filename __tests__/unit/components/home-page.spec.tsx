import * as React from 'react';
import { shallow } from 'enzyme';

import { HomePage } from '../../../src/components/';

describe ('HomePage', () => {
    it('renders correctly', () => {
        const homepage = shallow(<HomePage />);
      
        expect(homepage).toMatchSnapshot();
      });
});

