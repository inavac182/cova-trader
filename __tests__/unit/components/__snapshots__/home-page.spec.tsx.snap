// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div
  className="page"
>
  <h1>
    Hello!
  </h1>
</div>
`;
